source/common/redis/command_splitter_impl.cc

```diff
@@ -21,43 +21,50 @@ RespValuePtr Utility::makeError(const std::string& error) {
   return response;
 }
 
-SplitRequestPtr AllParamsToOneServerCommandHandler::startRequest(const RespValue& request,
-                                                                 SplitCallbacks& callbacks) {
-  std::unique_ptr<SplitRequestImpl> request_handle(new SplitRequestImpl(callbacks));
-  request_handle->handle_ =
-      conn_pool_.makeRequest(request.asArray()[1].asString(), request, *request_handle);
-  if (!request_handle->handle_) {
-    callbacks.onResponse(Utility::makeError("no upstream host"));
+SplitRequestPtr SimpleRequest::create(ConnPool::Instance& conn_pool,
+                                      const RespValue& incoming_request,
+                                      SplitCallbacks& callbacks) {
+  std::unique_ptr<SimpleRequest> request_ptr =
+      std::unique_ptr<SimpleRequest>{new SimpleRequest(callbacks)};
+
+  request_ptr->handle_ = conn_pool.makeRequest(incoming_request.asArray()[1].asString(),
+                                               incoming_request, *request_ptr);
+  if (!request_ptr->handle_) {
+    request_ptr->callbacks_.onResponse(Utility::makeError("no upstream host"));
     return nullptr;
   }
 
-  return std::move(request_handle);
+  return std::move(request_ptr);
 }
 
-AllParamsToOneServerCommandHandler::SplitRequestImpl::~SplitRequestImpl() { ASSERT(!handle_); }
-
-void AllParamsToOneServerCommandHandler::SplitRequestImpl::cancel() {
-  handle_->cancel();
+void SimpleRequest::onResponse(RespValuePtr&& response) {
   handle_ = nullptr;
+  callbacks_.onResponse(std::move(response));
 }
 
-void AllParamsToOneServerCommandHandler::SplitRequestImpl::onResponse(RespValuePtr&& response) {
+void SimpleRequest::onFailure() {
   handle_ = nullptr;
-  ENVOY_LOG(debug, "redis: response: '{}'", response->toString());
-  callbacks_.onResponse(std::move(response));
+  callbacks_.onResponse(Utility::makeError("upstream failure"));
 }
 
-void AllParamsToOneServerCommandHandler::SplitRequestImpl::onFailure() {
+void SimpleRequest::cancel() {
+  handle_->cancel();
   handle_ = nullptr;
-  callbacks_.onResponse(Utility::makeError("upstream failure"));
 }
 
-SplitRequestPtr MGETCommandHandler::startRequest(const RespValue& request,
-                                                 SplitCallbacks& callbacks) {
-  std::unique_ptr<SplitRequestImpl> request_handle(
-      new SplitRequestImpl(callbacks, request.asArray().size() - 1));
+SplitRequestPtr MGETRequest::create(ConnPool::Instance& conn_pool,
+                                    const RespValue& incoming_request, SplitCallbacks& callbacks) {
+  std::unique_ptr<MGETRequest> request_ptr =
+      std::unique_ptr<MGETRequest>{new MGETRequest(callbacks)};
```

Comment:
Here too, can you just do:

`std::unique_ptr<MGETRequest> request_ptr{new MGETRequest(callbacks)}`